packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -300,8 +304,36 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
 
   private void handleChoosePictureResult(int resultCode, Intent data) {
     if (resultCode == Activity.RESULT_OK && data != null) {
-      String path = fileUtils.getPathFromUri(activity, data.getData());
-      handleResult(path);
+        //String path = fileUtils.getPathFromUri(activity, data.getData());
+
+        //creating a temp file from uri to allow reading pictures from remote sources
+        try {
+            InputStream inputStream = null;
+            OutputStream output = null;
+            try {
+                inputStream = activity.getContentResolver().openInputStream(data.getData());
+                File file = new File(activity.getCacheDir(), "cachedImage.jpg");
+                output = new FileOutputStream(file);
+                if (inputStream != null) {
+                    byte[] buffer = new byte[4 * 1024]; // or other buffer size
```

Comment:
Please avoid comments that convey only uncertainty. There is nothing inherently wrong with a 4K buffer.